examples/common/foo_library/CMakeLists.txt

```diff
@@ -1,3 +1,7 @@
+if(DEFINED OPENTELEMETRY_BUILD_DLL)
+  add_definitions(-DOPENTELEMETRY_BUILD_IMPORT_DLL)
+endif()
```

Comment:
This is correcting global scope pollution caused by the top level CMakeList.txt,
see earlier comments.

examples are not part of the opentelemetry-cpp library, so compiling examples should never see OPENTELEMETRY_BUILD_DDL defined.

Reply:
If example sees `OPENTELEMETRY_BUILD_DLL`, it knows it is going to reference opentelemetry as DLL. `OPENTELEMETRY_BUILD_DLL` implies opentelemetry should be in DLL form instead of compiling any current code as DLL.